### PR TITLE
Standardize ParlAI script access

### PR DESCRIPTION
**Pull request by vedantpuri**
**Patch description**
Fix for one part of #2713 

**Testing steps**
Ensure CI doesn't break. Try checking ```--help``` works as intended for the upgraded scripts.

**Logs**
Example ```--help``` output for
```
 python -m parlai.scripts.interactive_web --help
```
```
usage: interactive_web.py [-h] [-o INIT_OPT] [-v] [-t TASK]
                          [-dt {train,train:stream,train:ordered,train:ordered:stream,train:stream:ordered,train:evalmode,train:evalmode:stream,train:evalmode:ordered,train:evalmode:ordered:stream,train:evalmode:stream:ordered,valid,valid:stream,test,test:stream}]
                          [-nt NUMTHREADS] [-bs BATCHSIZE]
                          [-dynb {batchsort,full,None}] [-dp DATAPATH]
                          [-m MODEL] [-mf MODEL_FILE] [-im INIT_MODEL]
                          [-d DISPLAY_EXAMPLES]
                          [--display-prettify DISPLAY_PRETTIFY]
                          [--display-ignore-fields DISPLAY_IGNORE_FIELDS]
                          [-it INTERACTIVE_TASK]
                          [--save-world-logs SAVE_WORLD_LOGS]
                          [-fixedCands LOCAL_HUMAN_CANDIDATES_FILE]
                          [--single-turn SINGLE_TURN] [--port PORT]
                          [--host HOST]

Interactive chat with a model

optional arguments:
  -h, --help
      show this help message and exit
  -d, --display-examples DISPLAY_EXAMPLES
  --display-prettify DISPLAY_PRETTIFY
      Set to use a prettytable when displaying examples with text candidates
      (default: False)
  --display-ignore-fields DISPLAY_IGNORE_FIELDS
      Do not display these fields (default: label_candidates,text_candidates)
  -it, --interactive-task INTERACTIVE_TASK
      Create interactive version of task (default: True)
  --save-world-logs SAVE_WORLD_LOGS
      Saves a jsonl file containing all of the task examples and model replies.
      Must also specify --report-filename. (default: False)
  --port PORT
      Port to listen on. (default: 8080)
  --host HOST
      Host from which allow requests, use 0.0.0.0 to allow all IPs (default:
      localhost)

Main ParlAI Arguments:
  -o, --init-opt INIT_OPT
      Path to json file of options. Note: Further Command-line arguments
      override file-based options. (default: None)
  -v, --show-advanced-args
      Show hidden command line options (advanced users only) (default: False)
  -t, --task TASK
      ParlAI task(s), e.g. "babi:Task1" or "babi,cbt" (default: interactive)
  -dt, --datatype {train,train:stream,train:ordered,train:ordered:stream,train:stream:ordered,train:evalmode,train:evalmode:stream,train:evalmode:ordered,train:evalmode:ordered:stream,train:evalmode:stream:ordered,valid,valid:stream,test,test:stream}
      choose from: train, train:ordered, valid, test. to stream data add
      ":stream" to any option (e.g., train:stream). by default: train is random
      with replacement, valid is ordered, test is ordered. (default: train)
  -nt, --numthreads NUMTHREADS
      number of threads. Used for hogwild if batchsize is 1, else for number of
      threads in threadpool loading, (default: 1)
  -bs, --batchsize BATCHSIZE
      batch size for minibatch training schemes (default: 1)
  -dynb, --dynamic-batching {batchsort,full,None}
      Use dynamic batching (default: None)
  -dp, --datapath DATAPATH
      path to datasets, defaults to {parlai_dir}/data (default: None)

ParlAI Model Arguments:
  -m, --model MODEL
      the model class name. can match parlai/agents/<model> for agents in that
      directory, or can provide a fully specified module for `from X import Y`
      via `-m X:Y` (e.g. `-m parlai.agents.seq2seq.seq2seq:Seq2SeqAgent`)
      (default: None)
  -mf, --model-file MODEL_FILE
      model file name for loading and saving models (default: None)
  -im, --init-model INIT_MODEL
      load model weights and dict from this file (default: None)

Local Human Arguments:
  -fixedCands, --local-human-candidates-file LOCAL_HUMAN_CANDIDATES_FILE
      File of label_candidates to send to other agent (default: None)
  --single-turn SINGLE_TURN
      If on, assumes single turn episodes. (default: False)

```

**Other information**
NA

**Data tests (if applicable)**
NA

### CODE REVIEW

**Comment by vedantpuri** [comment]
@stephenroller Any other ways to test this ? I don't think CI has tests for these scripts but I could be wrong

**Comment by stephenroller** [comment]
I think I’d be satisfied with —help producing the same results for all of them.

File: parlai/scripts/convert_data_to_parlai_format.py [hunk 1]
```diff
@@ -64,41 +65,47 @@ def dump_data(opt):
...
-class ConvertDataToParlaiFormat:
+def setup_args():
+    # Get command line arguments
+    parser = ParlaiParser()
+    parser.add_argument(
+        '-n',
+        '--num-examples',
+        default=-1,
+        type=int,
+        help='Total number of exs to convert, -1 to convert \
```
Comment (stephenroller) [inline parlai/scripts/convert_data_to_parlai_format.py:76]: nit: while you're hree, can you clean up these strings (make them multi-line strings like you see in other places).

**Comment by vedantpuri** [comment]
Unsure why actions didn't run on previous commit

File: parlai/scripts/distributed_train.py [hunk 2]
```diff
@@ -52,8 +53,10 @@ def main():
     parser = single_train.setup_args()
     parser.add_distributed_training_args()
     parser.add_argument('--port', type=int, default=61337, help='TCP port number')
-    opt = parser.parse_args(print_args=(os.environ['SLURM_PROCID'] == '0'))
+    return parser
 
+
+def dist_train(opt):
```
Comment (stephenroller) [inline parlai/scripts/distributed_train.py:51]: the node_list lint error is a bug

Comment (vedantpuri) [reply parlai/scripts/distributed_train.py:51]: I see, regardless, I am not sure why the lint and 3 other tests didn't run on latest commit.

Does the rest of the PR look good ?

Comment (stephenroller) [reply parlai/scripts/distributed_train.py:51]: man what is wrong with your account and CI...

You can run `bash autoformat.sh` to get lint locally

Comment (stephenroller) [reply parlai/scripts/distributed_train.py:51]: spot check they look fine but lint was definitely finding bugs from copy paste failures

Comment (vedantpuri) [reply parlai/scripts/distributed_train.py:51]: Output from ```bash autoformat.sh```

```
All done! ✨ 🍰 ✨
18 files left unchanged.
```

**Comment by vedantpuri** [comment]
Another commit solved it I guess

File: parlai/scripts/distributed_train.py [hunk 3]
```diff
@@ -84,5 +79,20 @@ def main():
...
+        return setup_args()
+
+    def run(self):
+        return dist_train(self.opt)
+
+
 if __name__ == '__main__':
-    main()
+    # double check we're using SLURM
+    node_list = os.environ.get('SLURM_JOB_NODELIST')
```
Comment (stephenroller) [inline parlai/scripts/distributed_train.py:93]: could this bit be moved into the DistributedTrain.run? I want it always checked.

File: parlai/scripts/distributed_train.py [hunk 4]
```diff
@@ -84,5 +79,20 @@ def main():
...
+    if node_list is None:
+        raise RuntimeError(
+            'Does not appear to be in a SLURM environment. You should not call this script directly; see launch_distributed.py'
+        )
+    DistributedTrain.main(print_args=(os.environ['SLURM_PROCID'] == '0'))
```
Comment (stephenroller) [inline parlai/scripts/distributed_train.py:98]: i think this is a bug. main shouldn't take print_args.

**Comment by stephenroller** [comment]
Okay, let's go!

**Pull request merged**